concord-server/server/src/test/java/com/rabidgremlin/concord/resources/LabelsResourceTest.java

```diff
@@ -18,58 +18,100 @@
     MockitoAnnotations.initMocks(this);
-    labelsResource = new LabelsResource(labelsDaoMock);
 
-    Label label1 = new Label();
-    label1.setLabel("OrderTaxi");
-    label1.setShortDescription("Order Taxi");
-    label1.setLongDescription("Where would you like your taxi to be sent ?");
+    setUpDatabase();
 
-    labels.add(label1);
+    labelsResource = createLabelsResource();
   }
 
   @Test
   public void canUploadCSVOfLabels()
   {
+    // given this list of labels
+    List<Label> labels = new ArrayList<>();
+    Label label1 = new Label();
+    label1.setLabel("OrderTaxi");
+    label1.setShortDescription("Order Taxi");
+    label1.setLongDescription("Where would you like your taxi to be sent ?");
+    labels.add(label1);
+
+    // when we call upload csv with list of labels
     Response response = labelsResource.uploadCsv(callerMock, labels);
 
-    verify(labelsDaoMock, times(1)).upsert(anyList());
+    // then it is successful
     assertThat(response, instanceOf(Response.class));
     assertEquals(200, response.getStatus());
     assertEquals("OK", response.getStatusInfo().toString());
+
+    // then the list of labels is stored in the db
+    List<Map<String, Object>> dbLabels = jdbi.withHandle(handle -> handle.createQuery("SELECT * FROM labels")
+        .mapToMap()
+        .list());
+
+    assertEquals(dbLabels.size(), 1); // should have one row
+    Map<String, Object> dbLabel = dbLabels.get(0);
+    assertEquals(dbLabel.size(), 3); // three columns
+    assertEquals("OrderTaxi", dbLabel.get("label"));
+    assertEquals("Order Taxi", dbLabel.get("shortdescription"));
+    assertEquals("Where would you like your taxi to be sent ?", dbLabel.get("longdescription"));
   }
 
   @Test
   public void canReturnAllLabels()
   {
+    // given this list of labels stored in the database
+    List<Label> labels = new ArrayList<>();
+    Label label1 = new Label();
+    label1.setLabel("OrderTaxi");
+    label1.setShortDescription("Order Taxi");
+    label1.setLongDescription("Where would you like your taxi to be sent ?");
+    labels.add(label1);
     labels.add(new Label("CancelTaxi", "Cancel Taxi", "Cancel my taxi"));
     labels.add(new Label("WhereTaxi", "Where taxi", "Where is my taxi"));
+    Response response = labelsResource.uploadCsv(callerMock, labels);
+    assertThat(response, instanceOf(Response.class));
+    assertEquals(200, response.getStatus());
 
-    when(labelsDaoMock.getLabels()).thenReturn(labels);
+    // when we retrieve the list of labels
 
-    Response response = labelsResource.getLabels(callerMock);
+    response = labelsResource.getLabels(callerMock);
 
+    // then we get backthe list of labels we expect
+    // TODO use sql query here
     assertThat(response, instanceOf(Response.class));
     assertEquals(200, response.getStatus());
     assertEquals("OK", response.getStatusInfo().toString());
-    assertEquals(response.getEntity(), labels);
+    assertEquals(((List) response.getEntity()).size(), 3);
```

Comment:
expected/actual is in wrong order in some cases

Reply:
```assertEquals``` does it the other way to ```assertThat``` apparently :)

Reply:
Would be better if we switch to [assertJ](http://joel-costigliola.github.io/assertj/) , which is more fluent and better in many ways.

Reply:
Yep assertJ would make sense. Also has some nice ways of validating maps and lists so will be much easier to validate SQL responses and REST responses.